src/back/msl/writer.rs

```diff
@@ -311,6 +311,32 @@ fn separate(need_separator: bool) -> &'static str {
     }
 }
 
+fn should_pack_struct_member(
+    members: &[crate::StructMember],
+    span: u32,
+    index: usize,
+    module: &crate::Module,
+) -> bool {
```

Comment:
maybe it can return `Option<(crate::VectorSize, crate::ScalarKind)>` for convenience?